### PR TITLE
feat: migrate Flow.cs to Flow.ts

### PR DESCRIPTION
**Pull request by ephread**
This is a completely new class.

Nothing particularly stands out about that file.

### CODE REVIEW

File: src/engine/Flow.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    let name = arguments[0] as string;
+    let story = arguments[1] as Story;
+
+    this.name = name;
+    this.callStack = new CallStack(story);
+
+    if (arguments[3]) {
+      let jObject = arguments[3] as Record<string, any>;
+
+      this.outputStream = JsonSerialisation.JArrayToRuntimeObjList(
```
Comment (y-lohse) [inline src/engine/Flow.ts:27]: I'm not 100% sure, but it seems like in this signature, we also need to do `this.callStack.SetJsonToken ((Dictionary < string, object > )jObject ["callstack"], story);` (l. 21 of Flow.cs https://github.com/inkle/ink/compare/0.9.0%E2%80%A6v1.0.0#diff-8cbd10a71465fea0e316355215ad5788ea62c3094e5b1e27d8e99eff1c1ee9f1R21). I think we may be missing that here?

**Review by y-lohse — COMMENTED** [review]
One question, other than that LGTM!

File: src/engine/Flow.ts (hunk 1 above)
Comment (ephread) [reply src/engine/Flow.ts:27]: Oh yeah, you're right, good catch! I'll fix that.

**Review by y-lohse — APPROVED** [review]
Sweet, thanks!

**Pull request merged**